### PR TITLE
[PY-400][external] Move Meta Item & ItemQuery to folder

### PR DESCRIPTION
**Pull request by JBWilkie**
# Problem
Currently no Meta methods for moving items between folders

# Solution
Introduced functions to do this for Meta Items & ItemQuery objects

# Changelog
Meta methods to move Items & ItemQuery objects between folders

### CODE REVIEW

File: darwin/future/core/items/move_items_to_folder.py [hunk 1]
```diff
@@ -44,7 +44,8 @@ def move_list_of_items_to_folder(
             if isinstance(dataset_ids, list)
             else [dataset_ids],
             **filters,
-        }
+        },
+        "path": path,
```
Comment (JBWilkie) [inline darwin/future/core/items/move_items_to_folder.py:48]: In PY-560 refactoring, I accidentally removed path & priority from their core methods respectively

Added them back

File: darwin/future/core/items/set_item_priority.py [hunk 2]
```diff
@@ -42,7 +42,8 @@ def set_item_priority(
             if isinstance(dataset_ids, list)
             else [dataset_ids],
             **filters,
-        }
+        },
+        "priority": priority,
```
Comment (JBWilkie) [inline darwin/future/core/items/set_item_priority.py:46]: Same as above

File: darwin/future/tests/meta/objects/test_itemmeta.py [hunk 3]
```diff
@@ -44,3 +44,28 @@ def test_delete(item: Item) -> None:
             json={},
         )
         item.delete()
+
+
+def test_move_to_folder(item: Item) -> None:
+    with responses.RequestsMock() as rsps:
```
Comment (owencjones) [inline darwin/future/tests/meta/objects/test_itemmeta.py:50]: Consider testing action result, and also adding sad path.

File: darwin/future/tests/meta/queries/test_item.py [hunk 4]
```diff
@@ -70,3 +70,40 @@ def test_delete(
             json={},
         )
         item_query.delete()
+
+
+def test_move_tofolderS(
```
Comment (owencjones) [inline darwin/future/tests/meta/queries/test_item.py:75]: Typo in def.  
```suggestion
def test_move_to_folders(
```

File: darwin/future/tests/meta/queries/test_item.py [hunk 5]
```diff
@@ -70,3 +70,40 @@ def test_delete(
...
+                            "item_ids": [str(item.id) for item in items],
+                            "dataset_ids": [dataset_id],
+                        },
+                        "path": path,
+                    }
+                )
+            ],
+            json={},
+        )
+        item_query.move_to_folder(path)
```
Comment (owencjones) [inline darwin/future/tests/meta/queries/test_item.py:109]: Consider a sad path.

**Review by owencjones — APPROVED** [review]
Comments to review, but not going to block a merge.

File: darwin/future/tests/meta/objects/test_itemmeta.py (hunk 3 above)
Comment (JBWilkie) [reply darwin/future/tests/meta/objects/test_itemmeta.py:52]: Good shout on the sad path, have added - On the subject of testing the action result, how would you approach this?

In this case, `move_to_folder()` doesn't return anything to check against, and the request is mocked. What can we check against to verify that the action had the intended effect?

**Pull request merged**